packages/flutter_test/lib/src/widget_tester.dart

```diff
@@ -59,6 +59,17 @@ export 'package:test_api/test_api.dart' hide
 /// Signature for callback to [testWidgets] and [benchmarkWidgets].
 typedef WidgetTesterCallback = Future<void> Function(WidgetTester widgetTester);
 
+// [Iterable.lastWhere] but returns null if not found.
+T? _lastWhereOrNull<T>(Iterable<T> list, bool Function(T) condition) {
+  try {
```

Comment:
This is using try/catch for actual control flow, let's not do that :)

Reply:
I would, but somehow normal cast with `lastWhere` throws strange null assertions. I guess I'll write a manual traversal then.